contrib/run_task.sh

```diff
@@ -205,45 +269,54 @@ build_docs_with_stable_toolchain() {
     RUSTDOCFLAGS="-D warnings" $cargo doc --all-features
 }
 
+# Bench only works with a non-stable toolchain (nightly, beta).
+do_bench() {
+    for crate in bitcoin hashes; do
+        pushd "$REPO_DIR/$crate" > /dev/null
+        RUSTFLAGS='--cfg=bench' cargo bench
+        popd > /dev/null
+    done
+}
+
+# This is only relevant for hashes.
+do_schemars() {
+    pushd "$REPO_DIR/hashes/extended_tests/schemars" > /dev/null
+    cargo test
+    popd > /dev/null
+}
+
 do_wasm() {
+    pushd "$REPO_DIR/hashes" > /dev/null
+
     clang --version &&
-	CARGO_TARGET_DIR=wasm cargo install --force wasm-pack &&
-	printf '\n[target.wasm32-unknown-unknown.dev-dependencies]\nwasm-bindgen-test = "0.3"\n' >> Cargo.toml &&
-	printf '\n[lib]\ncrate-type = ["cdylib", "rlib"]\n' >> Cargo.toml &&
-	CC=clang-9 wasm-pack build &&
-	CC=clang-9 wasm-pack test --node;
+        CARGO_TARGET_DIR=wasm cargo install --force wasm-pack &&
+        printf '\n' >> Cargo.toml &&
+        printf '[target.wasm32-unknown-unknown.dev-dependencies]\n' >> Cargo.toml &&
+        printf 'wasm-bindgen-test = "0.3.41"\n' >> Cargo.toml &&
+        CC=clang-9 wasm-pack build &&
+        CC=clang-9 wasm-pack test --node;
+
+    popd > /dev/null
 }
 
 do_asan() {
     cargo clean
     CC='clang -fsanitize=address -fno-omit-frame-pointer'                                        \
       RUSTFLAGS='-Zsanitizer=address -Clinker=clang -Cforce-frame-pointers=yes'                    \
       ASAN_OPTIONS='detect_leaks=1 detect_invalid_pointer_pairs=1 detect_stack_use_after_return=1' \
-      cargo test --lib --no-default-features --features="$ASAN_FEATURES" -Zbuild-std --target x86_64-unknown-linux-gnu
+      cargo test --lib --no-default-features --features="${ASAN_FEATURES:-}" -Zbuild-std --target x86_64-unknown-linux-gnu
```

Comment:
In d457ed8d54d72fdb7e09b69d38527f864f022a75:

I guess the chatbot suggested this? I see no value in it, it's just extra line noise.

Reply:
ha! I should have picked the bot up on that one. I can't decide who's better the bot or a first year uni student.

Reply:
Double ha! I take all those rude words back, the ~first year student~ bot was correct. Because we use `set -euox pipefail` we have to set the var to default otherwise we get a variable not set error.

Reply:
Ok, cool, good to know. (This may have been the cause of the examples "not being run" without destring-ifying the array.)

Reply:
Maybe yes, since `ASAN_FEATURES` is only set in the `hashes` `contrib/test_vars.sh` file and not the others. This thread made me realize I'd written in another bug - I wasn't sourcing `testing_vars.sh` at all in the asan test.